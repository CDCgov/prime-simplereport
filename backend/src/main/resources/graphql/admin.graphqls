# Queries and mutations that are used only by site admins
# All queries and mutations in this file implicitly require site admin access,
# which is enforced in the API not in the schema validator.
extend type Query {
  organizations(identityVerified: Boolean): [Organization!]!
  pendingOrganizations: [PendingOrganization!]!
}
extend type Mutation {
  uploadPatients(patientList: Upload!): String
  resendToReportStream(testEventIds: [ID!]!): Boolean
  createDeviceType(input: CreateDeviceType!): DeviceType
  updateDeviceType(input: UpdateDeviceType!): DeviceType
  addUser( # not actually used in the admin console (currently)
    name: NameInput
    firstName: String
    middleName: String
    lastName: String
    suffix: String
    email: String!
    organizationExternalId: String!
    role: Role!
  ): User
  createOrganization(
    name: String!
    type: String!
    externalId: String!
    testingFacilityName: String!
    cliaNumber: String
    street: String!
    streetTwo: String
    city: String
    county: String
      @deprecated(reason: "county is derived from smarty streets lookup")
    state: String!
    zipCode: String!
    phone: String
    email: String
    orderingProviderName: NameInput
    orderingProviderFirstName: String
    orderingProviderMiddleName: String
    orderingProviderLastName: String
    orderingProviderSuffix: String
    orderingProviderNPI: String
    orderingProviderStreet: String
    orderingProviderStreetTwo: String
    orderingProviderCity: String
    orderingProviderCounty: String
      @deprecated(reason: "county is derived from smarty streets lookup")
    orderingProviderState: String
    orderingProviderZipCode: String
    orderingProviderPhone: String
    deviceTypes: [String]!
    defaultDevice: String!
    adminName: NameInput
    adminFirstName: String
    adminMiddleName: String
    adminLastName: String
    adminSuffix: String
    adminEmail: String!
  ): Organization
  setOrganizationIdentityVerified(
    externalId: String!
    verified: Boolean!
  ): Boolean
  editPendingOrganization(
    orgExternalId: String!
    name: String
    adminFirstName: String
    adminLastName: String
    adminEmail: String
    adminPhone: String
  ): String
  markPendingOrganizationAsDeleted(
    orgExternalId: String!
    deleted: Boolean!
  ): String
  createOrganizationRegistrationLink(
    organizationExternalId: String!
    link: String!
  ): String
  createFacilityRegistrationLink(
    organizationExternalId: String!
    facilityId: ID!
    link: String!
  ): String
  updateRegistrationLink(link: String!, newLink: String!): String
  setRegistrationLinkIsDeleted(link: String, deleted: Boolean!): String
  adminUpdateOrganization(name: String!, type: String!): String
  markFacilityAsDeleted(facilityId: ID!, deleted: Boolean!): String
}
