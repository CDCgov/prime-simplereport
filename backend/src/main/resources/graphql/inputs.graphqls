input ProviderInput {
    fullName:NameInput
    firstName: String
    middleName: String
    lastName: String
    suffix: String
    npi: String
    street:String
    streetTwo:String
    city: String
    state: String
    zipCode: String
    phone: String
}