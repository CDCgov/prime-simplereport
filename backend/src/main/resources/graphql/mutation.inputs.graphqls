# Contains input definitions for mutations
input ProviderInput {
    firstName: String
    middleName: String
    lastName: String
    suffix: String
    npi: String
    street:String
    streetTwo:String
    city: String
    county: String
    state: String
    zipCode: String
    phone: String
}

input FacilityAddressInput{
    street: String!
    streetTwo: String
    city: String
    state: String!
    zipCode: String!
}

input UpdateFacilityInput {
    facilityId: ID!
    facilityName: String!
    cliaNumber: String
    address: FacilityAddressInput!
    phone: String
    email: String
    orderingProvider: ProviderInput
    deviceIds: [ID]!
}

input AddFacilityInput {
    facilityName: String!
    cliaNumber: String
    address: FacilityAddressInput!
    phone: String
    email: String
    orderingProvider: ProviderInput
    deviceIds: [ID]!
}