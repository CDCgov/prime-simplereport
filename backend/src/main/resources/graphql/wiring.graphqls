# Directives, enumerations, and utility types (e.g. structured name/address information, input types, etc.)
# go here, so that they can be easily referred to side by side with the more complex types

# Requires that the requester have a certain permission level.
#
# If applied to a field, the requester must meet the required permission level in order to request
# or traverse the targeted field. If the requirement is not met, `null` will be supplied for the
# targeted field and an error returned.
#
# If applied to an object, the request must meet the required permission level in order to request
# any fields of the object. If any child fields are themselves targeted with this directive, the
# requester must satisfy both object- and field-level directives to request or traverse said fields.
#
# If applied to an argument, the requester must meet the required permission level in order to
# supply a non-default value for the targeted argument; otherwise, an error will be returned and the
# query or mutation housing the argument will not be executed.
directive @requiredPermissions(
  anyOf: [String!] # if the requester lacks any of these permissions, deny
  allOf: [String!] # if the requester lacks all of these permissions, deny
) on FIELD_DEFINITION | OBJECT | ARGUMENT_DEFINITION

# These directives come from the graphql-java-extended-validation package. This package offers a
# number of additional directives (@Range, @Digits, etc.) that can be pulled in and enforced
# automatically by graphql-java-extended-validation once they are defined in this file.
# Cf. https://github.com/graphql-java/graphql-java-extended-validation for a complete list of
# directives that can be used.
directive @Size(
  min: Int!
  max: Int!
  message: String = "graphql.validation.Size.message"
) on ARGUMENT_DEFINITION
directive @Pattern(
  regexp: String!
  message: String = "graphql.validation.Pattern.message"
) on ARGUMENT_DEFINITION

# custom scalar gov.cdc.usds.simplereport.config.scalars.datetime.DateTimeScalar
scalar DateTime
# custom scalar gov.cdc.usds.simplereport.api.FlexibleDateScalar
scalar LocalDate

enum UploadStatus {
  PENDING
  SUCCESS
  FAILURE
}

enum ResultValue {
  POSITIVE
  NEGATIVE
  UNDETERMINED
}

enum UserPermission {
  READ_PATIENT_LIST
  READ_ARCHIVED_PATIENT_LIST
  SEARCH_PATIENTS
  READ_RESULT_LIST
  EDIT_PATIENT
  ARCHIVE_PATIENT
  EDIT_FACILITY
  EDIT_ORGANIZATION
  MANAGE_USERS
  START_TEST
  UPDATE_TEST
  SUBMIT_TEST
  ACCESS_ALL_FACILITIES
  UPLOAD_RESULTS_SPREADSHEET
  VIEW_ARCHIVED_FACILITIES
}

enum Role {
  ENTRY_ONLY
  USER
  ADMIN
  TEST_RESULT_UPLOAD_USER
}

enum TestCorrectionStatus {
  ORIGINAL
  CORRECTED
  REMOVED
}

enum TestResultDeliveryPreference {
  SMS
  EMAIL
  ALL
  NONE
}

enum PhoneType {
  MOBILE
  LANDLINE
}

type NameInfo {
  firstName: String
  middleName: String
  lastName: String!
  suffix: String
}

type AddressInfo {
  streetOne: String
  streetTwo: String
  city: String
  county: String
  state: String
  postalCode: String
}

input NameInput {
  firstName: String
  middleName: String
  lastName: String
  suffix: String
}
