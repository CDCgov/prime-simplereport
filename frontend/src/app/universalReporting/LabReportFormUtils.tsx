import {
  Condition,
  FacilityReportInput,
  PatientReportInput,
  ProviderReportInput,
  ResultScaleType,
  Specimen,
  SpecimenBodySite,
  SpecimenInput,
} from "../../generated/graphql";
import { RadioGroupOptions } from "../commonComponents/RadioGroup";
import { TEST_RESULTS_SNOMED } from "../testResults/constants";

export const defaultPatientReportInputState: PatientReportInput = {
  city: "",
  country: "USA",
  county: "",
  dateOfBirth: "",
  email: "",
  ethnicity: "",
  firstName: "",
  lastName: "",
  middleName: "",
  patientId: "",
  phone: "",
  race: "",
  sex: "",
  state: "",
  street: "",
  streetTwo: "",
  tribalAffiliation: "",
  zipCode: "",
};

export const defaultProviderReportInputState: ProviderReportInput = {
  city: "",
  county: "",
  email: "",
  firstName: "Count",
  lastName: "Dracula",
  middleName: "",
  npi: "0000000000",
  phone: "(555) 555-5555",
  state: "",
  street: "",
  streetTwo: "",
  suffix: "",
  zipCode: "",
};

export const defaultFacilityReportInputState: FacilityReportInput = {
  city: "Jefferson City",
  clia: "00D0000000",
  county: "Cole",
  email: "",
  name: "Dracula",
  phone: "(555) 555-5555",
  state: "MO",
  street: "201 W Capitol Ave",
  streetTwo: "",
  zipCode: "65101-6809",
};

export const defaultSpecimenReportInputState: SpecimenInput = {
  snomedTypeCode: "",
  collectionDate: "",
  receivedDate: "",
  collectionBodySiteCode: "",
  collectionBodySiteName: "",
};

export const mapScaleDisplayToResultScaleType = (scaleDisplay: string) => {
  switch (scaleDisplay) {
    case "Nom":
      return ResultScaleType.Nominal;
    case "Ord":
      return ResultScaleType.Ordinal;
    case "Qn":
      return ResultScaleType.Quantitative;
    default:
      return ResultScaleType.Ordinal;
  }
};

export const ordinalResultOptions: RadioGroupOptions<string> = [
  {
    value: TEST_RESULTS_SNOMED.POSITIVE,
    label: `Positive (+)`,
  },
  {
    value: TEST_RESULTS_SNOMED.NEGATIVE,
    label: `Negative (-)`,
  },
  {
    value: TEST_RESULTS_SNOMED.UNDETERMINED,
    label: `Undetermined`,
  },
];

export const buildSpecimenOptionList = (specimens: Specimen[]) => {
  const options = specimens.map((specimen) => {
    return {
      value: specimen.snomedCode,
      label: `${specimen.snomedDisplay} - ${specimen.snomedCode}`,
    };
  });
  options.sort((a, b) => a.label.localeCompare(b.label));
  return options;
};

export const ResultScaleTypeOptions: {
  label: string;
  value: ResultScaleType;
}[] = [
  {
    label: ResultScaleType.Ordinal,
    value: ResultScaleType.Ordinal,
  },
  {
    label: ResultScaleType.Quantitative,
    value: ResultScaleType.Quantitative,
  },
  {
    label: ResultScaleType.Nominal,
    value: ResultScaleType.Nominal,
  },
];

export const buildConditionsOptionList = (conditions: Condition[]) => {
  const options = conditions.map((condition) => {
    const fullConditiondisplay = condition.snomedName
      ? `${condition.display}  -  ${condition.snomedName}  -  ${condition.code}`
      : `${condition.display}  -  ${condition.code}`;
    return {
      value: condition.code,
      label: fullConditiondisplay,
    };
  });
  options.sort((a, b) => a.label.localeCompare(b.label));
  return options;
};

export const buildBodySiteOptionsList = (
  specimenBodySites: SpecimenBodySite[]
) => {
  if (specimenBodySites.length === 0) {
    return [
      {
        value: "",
        label: "No collection location available",
      },
    ];
  }
  const options = specimenBodySites.map((bodySite) => {
    return {
      value: bodySite.snomedSiteCode,
      label: bodySite.snomedSiteDisplay,
    };
  });
  options.sort((a, b) => a.label.localeCompare(b.label));
  return options;
};
