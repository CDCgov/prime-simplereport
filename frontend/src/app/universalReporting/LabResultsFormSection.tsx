import React, { Dispatch, useState } from "react";
import { ComboBox } from "@trussworks/react-uswds";

import {
  Lab,
  SpecimenInput,
  TestDetailsInput,
  useGetConditionsQuery,
  useGetLabsByConditionsLazyQuery,
  useGetSpecimensByLoincLazyQuery,
} from "../../generated/graphql";

import "./universalReporting.scss";
import TestOrderFormSubsection from "./TestOrderFormSubsection";
import SpecimenFormSubsection from "./SpecimenFormSubsection";
import TestDetailFormSubsection from "./TestDetailFormSubsection";
import {
  buildConditionsOptionList,
  defaultSpecimenReportInputState,
  mapScaleDisplayToResultScaleType,
} from "./LabReportFormUtils";

type LabResultsFormSectionProps = {
  specimen: SpecimenInput;
  setSpecimen: Dispatch<SpecimenInput>;
  testDetailList: TestDetailsInput[];
  setTestDetailList: Dispatch<TestDetailsInput[]>;
};

const LabResultsFormSection = ({
  specimen,
  setSpecimen,
  testDetailList,
  setTestDetailList,
}: LabResultsFormSectionProps) => {
  const [selectedCondition, setSelectedCondition] = useState<string>("");
  const [testOrderLoinc, setTestOrderLoinc] = useState<string>("");
  const [testOrderSearchString, setTestOrderSearchString] =
    useState<string>("");

  const { data: conditionsData, loading: conditionsLoading } =
    useGetConditionsQuery();
  const [getLabsByConditions, { data: labData, loading: labDataLoading }] =
    useGetLabsByConditionsLazyQuery();
  const [
    getSpecimensByLoinc,
    { data: specimenListData, loading: specimenListLoading },
  ] = useGetSpecimensByLoincLazyQuery();

  const conditionOptions = buildConditionsOptionList(
    conditionsData?.conditions ?? []
  );

  const updateCondition = async (selectedCondition: string) => {
    setSelectedCondition(selectedCondition);

    if (selectedCondition) {
      // until we implement multiplex testing, for now we are restricting the frontend to handling one condition at a time
      // even though the backend query can still support retrieving labs by multiple condition codes
      await getLabsByConditions({
        variables: {
          conditionCodes: [selectedCondition],
        },
      });
    } else {
      setTestDetailList([]);
      setTestOrderLoinc("");
      setTestOrderSearchString("");
      setSpecimen(defaultSpecimenReportInputState);
    }
  };

  const updateTestOrderLoinc = async (lab: Lab) => {
    const updatedList = [] as TestDetailsInput[];
    updatedList.push({
      condition: selectedCondition,
      testOrderLoinc: lab.code,
      testOrderDisplayName: lab.display,
      testPerformedLoinc: lab.code,
      testPerformedLoincLongCommonName: lab.longCommonName,
      resultType: mapScaleDisplayToResultScaleType(lab.scaleDisplay ?? ""),
      resultValue: "",
      resultDate: "",
      resultInterpretation: "",
    } as TestDetailsInput);
    setTestDetailList(updatedList);
    setTestOrderLoinc(lab.code);

    if (lab.systemCode) {
      const response = await getSpecimensByLoinc({
        variables: {
          loinc: lab.systemCode,
        },
      });
      const specimenData = response.data?.specimens ?? [];
      const sortedSpecimenData = specimenData.toSorted((a, b) =>
        a.snomedDisplay.localeCompare(b.snomedDisplay)
      );
      const sortedBodySiteList =
        sortedSpecimenData[0].bodySiteList?.toSorted((a, b) =>
          a.snomedSiteDisplay.localeCompare(b.snomedSiteDisplay)
        ) ?? [];

      setSpecimen({
        ...specimen,
        snomedTypeCode: sortedSpecimenData[0].snomedCode,
        snomedDisplayName: sortedSpecimenData[0].snomedDisplay,
        collectionBodySiteCode: sortedBodySiteList[0]?.snomedSiteCode ?? "",
        collectionBodySiteName: sortedBodySiteList[0]?.snomedSiteDisplay ?? "",
      } as SpecimenInput);
    } else {
      // currently filtering out labs with no system code on the backend
      console.error("No LOINC system code to look up specimen.");
    }
  };

  const updateTestDetails = (details: TestDetailsInput) => {
    let updatedList = [...testDetailList];
    updatedList = updatedList.filter(
      (x) => x.testPerformedLoinc !== details.testPerformedLoinc
    );
    updatedList = [...updatedList, details];
    setTestDetailList(updatedList);
  };

  return (
    <>
      <div className="grid-row grid-gap">
        <div className="grid-col-auto">
          <h2 className={"font-sans-lg"}>Condition Tested</h2>
        </div>
      </div>
      <div className="grid-row margin-bottom-2">
        <div className="grid-col-10">
          {conditionsLoading ? (
            <div>Loading condition list...</div>
          ) : (
            <>
              <label
                className="usa-legend margin-top-0"
                htmlFor="selected-condition"
              >
                Condition to report
              </label>
              <ComboBox
                id="selected-condition"
                name="selected-condition"
                options={conditionOptions}
                onChange={(e) => updateCondition(e ?? "")}
                defaultValue={selectedCondition}
                aria-required={true}
                className={"condition-combo-box"}
              />
            </>
          )}
        </div>
      </div>
      <TestOrderFormSubsection
        hasSelectedCondition={!!selectedCondition}
        labDataLoading={labDataLoading}
        labs={labData?.labs ?? []}
        testOrderLoinc={testOrderLoinc}
        updateTestOrderLoinc={updateTestOrderLoinc}
        testOrderSearchString={testOrderSearchString}
        setTestOrderSearchString={setTestOrderSearchString}
      />
      <SpecimenFormSubsection
        specimen={specimen}
        setSpecimen={setSpecimen}
        loading={specimenListLoading}
        isTestOrderSelected={testOrderLoinc.length > 0}
        specimenList={specimenListData?.specimens ?? []}
      />
      {testDetailList
        .sort((a, b) =>
          a.testPerformedLoinc.localeCompare(b.testPerformedLoinc)
        )
        .map((testDetails) => {
          return (
            <div
              className={"margin-top-2"}
              key={testDetails.testPerformedLoinc}
            >
              <TestDetailFormSubsection
                testDetails={testDetails}
                updateTestDetails={updateTestDetails}
              />
            </div>
          );
        })}
    </>
  );
};

export default LabResultsFormSection;
