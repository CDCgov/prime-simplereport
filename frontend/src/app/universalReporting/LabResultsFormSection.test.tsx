import configureStore, { MockStoreEnhanced } from "redux-mock-store";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/client/testing";
import { Provider } from "react-redux";

import { GetAllLabsDocument } from "../../generated/graphql";

import LabResultsFormSection from "./LabResultsFormSection";
import { defaultSpecimenReportInputState } from "./LabReportFormUtils";

describe("LabResultsFormSection", () => {
  let store: MockStoreEnhanced<unknown, {}>;
  const mockStore = configureStore([]);

  beforeEach(() => {
    jest.spyOn(Date, "now").mockReturnValue(new Date("2025-09-15").getTime());

    store = mockStore({
      organization: {
        name: "Organization Name",
      },
      facilities: [
        {
          id: "a1",
          name: "Fake Facility",
        },
      ],
      user: {
        permissions: [],
      },
    });
  });

  it("renders successfully after loading", async () => {
    const { container } = render(
      <MemoryRouter>
        <MockedProvider mocks={mocks}>
          <Provider store={store}>
            <LabResultsFormSection
              specimen={defaultSpecimenReportInputState}
              setSpecimen={() => {}}
              testDetailList={[]}
              setTestDetailList={() => {}}
            />
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(await screen.findByText("Loading test orders..."));

    expect(
      await screen.findByText("Choose a specimen type to enter test results")
    );
    expect(
      await screen.findByText(
        "Please select test order and specimen type above to see test result information"
      )
    );
    expect(container).toMatchSnapshot();
  });
});

const mocks = [
  {
    request: {
      query: GetAllLabsDocument,
    },
    result: {
      data: {
        labs: [
          {
            code: "101289-7",
            description: null,
            display:
              "SARS-CoV-2 (COVID-19) RNA [Presence] in Throat by NAA with non-probe detection",
            longCommonName:
              "SARS-CoV-2 (COVID-19) RNA [Presence] in Throat by NAA with non-probe detection",
            scaleCode: "LP7751-3",
            scaleDisplay: "Ord",
            systemCode: "LP7633-3",
            systemDisplay: "Thrt",
            answerList: "LL360-9",
            orderOrObservation: "Both",
            panel: false,
          },
          {
            code: "8023-4",
            description: null,
            display:
              "Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay",
            longCommonName:
              "Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay",
            scaleCode: "LP7751-3",
            scaleDisplay: "Ord",
            systemCode: "LP7567-3",
            systemDisplay: "Ser",
            answerList: "",
            orderOrObservation: "Both",
            panel: false,
          },
        ],
      },
    },
  },
];
