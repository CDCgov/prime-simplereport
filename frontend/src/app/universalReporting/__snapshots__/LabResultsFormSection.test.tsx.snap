// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabResultsFormSection renders successfully after loading 1`] = `
<div>
  <div
    class="grid-row"
  >
    <div
      class="grid-col-auto"
    >
      <h3
        class="margin-bottom-0 margin-top-1"
      >
        Test order information
      </h3>
      <p>
        Choose a test order to select specimen type
      </p>
    </div>
  </div>
  <div
    class="grid-row grid-gap"
  >
    <div
      class="grid-col-6 grid-col-mobile"
    >
      <label
        class="usa-legend margin-top-0"
        for="selected-lab"
      >
        Test order
      </label>
      <div
        class="usa-combo-box lab-combo-box"
        data-enhanced="true"
        data-testid="combo-box"
      >
        <select
          aria-hidden="true"
          class="usa-select usa-sr-only usa-combo-box__select"
          data-testid="combo-box-select"
          name="selected-lab"
          tabindex="-1"
        >
          <option
            value="8023-4"
          >
            Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay
          </option>
          <option
            value="101289-7"
          >
            SARS-CoV-2 (COVID-19) RNA [Presence] in Throat by NAA with non-probe detection
          </option>
        </select>
        <input
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-controls="selected-lab--list"
          aria-describedby="selected-lab--assistiveHint"
          aria-expanded="false"
          aria-owns="selected-lab--list"
          autocapitalize="off"
          autocomplete="off"
          class="usa-combo-box__input"
          data-testid="combo-box-input"
          id="selected-lab"
          role="combobox"
          type="text"
          value=""
        />
        <span
          class="usa-combo-box__clear-input__wrapper"
          tabindex="-1"
        >
          <button
            aria-label="Clear the select contents"
            class="usa-combo-box__clear-input"
            data-testid="combo-box-clear-button"
            hidden=""
            type="button"
          >
             
          </button>
        </span>
        <span
          class="usa-combo-box__input-button-separator"
        >
           
        </span>
        <span
          class="usa-combo-box__toggle-list__wrapper"
          tabindex="-1"
        >
          <button
            aria-label="Toggle the dropdown list"
            class="usa-combo-box__toggle-list"
            data-testid="combo-box-toggle"
            tabindex="-1"
            type="button"
          >
             
          </button>
        </span>
        <ul
          class="usa-combo-box__list"
          data-testid="combo-box-option-list"
          hidden=""
          id="selected-lab--list"
          role="listbox"
          tabindex="-1"
        >
          <li
            aria-posinset="1"
            aria-selected="false"
            aria-setsize="2"
            class="usa-combo-box__list-option"
            data-testid="combo-box-option-8023-4"
            data-value="8023-4"
            id="selected-lab--list--option-0"
            role="option"
            tabindex="-1"
            value="8023-4"
          >
            Saint Louis encephalitis virus Ab [Presence] in Serum by Immunoassay
          </li>
          <li
            aria-posinset="2"
            aria-selected="false"
            aria-setsize="2"
            class="usa-combo-box__list-option"
            data-testid="combo-box-option-101289-7"
            data-value="101289-7"
            id="selected-lab--list--option-1"
            role="option"
            tabindex="-1"
            value="101289-7"
          >
            SARS-CoV-2 (COVID-19) RNA [Presence] in Throat by NAA with non-probe detection
          </li>
        </ul>
        <div
          class="usa-combo-box__status usa-sr-only"
          role="status"
        />
        <span
          class="usa-sr-only"
          data-testid="combo-box-assistive-hint"
          id="selected-lab--assistiveHint"
        >
          When autocomplete results are available use up and down arrows to review
           and enter to select. Touch device users, explore by touch or with swipe
           gestures.
        </span>
      </div>
    </div>
  </div>
  <div
    class="grid-row margin-top-2"
  >
    <div
      class="grid-col-auto"
    >
      <h3
        class="margin-bottom-0 margin-top-5"
      >
        Specimen information
      </h3>
      <p
        class="margin-bottom-0"
      >
        Choose a specimen type to enter test results
      </p>
    </div>
  </div>
  <div
    class="grid-row grid-gap"
  >
    <div
      class="grid-col-6 grid-col-mobile"
    >
      <div
        class="usa-form-group prime-dropdown  card-dropdown"
      >
        <label
          class="usa-label"
          for="1"
        >
          Specimen type
        </label>
        <select
          aria-required="false"
          class="usa-select"
          id="1"
          name="specimen-type"
        >
          <option
            value=""
          />
        </select>
      </div>
    </div>
  </div>
  <div
    class="grid-row grid-gap"
  >
    <div
      class="grid-col-6 grid-col-mobile"
    >
      <div
        class="usa-form-group"
      >
        <label
          class="usa-label"
          for="2"
        >
          Collection date
        </label>
        <input
          aria-required="false"
          class="usa-input"
          id="2"
          max="2025-09-15"
          min="2020-01-01"
          name="specimen-collection-date"
          type="date"
          value="Invalid date"
        />
      </div>
    </div>
    <div
      class="grid-col-6 grid-col-mobile"
    >
      <div
        class="usa-form-group"
      >
        <label
          class="usa-label"
          for="3"
        >
          Collection time
        </label>
        <input
          aria-required="false"
          class="usa-input"
          id="3"
          name="specimen-collection-time"
          step="60"
          type="time"
          value="Invalid date"
        />
      </div>
    </div>
  </div>
  <div
    class="grid-row grid-gap"
  >
    <div
      class="grid-col-6 grid-col-mobile"
    >
      <div
        class="usa-form-group prime-dropdown  card-dropdown"
      >
        <label
          class="usa-label"
          for="4"
        >
          Specimen site (optional)
        </label>
        <select
          aria-required="false"
          class="usa-select"
          id="4"
          name="specimen-collection-body-site"
        >
          <option
            aria-selected="true"
            value=""
          >
            No collection location available
          </option>
        </select>
      </div>
    </div>
  </div>
  <div>
    <div
      class="grid-col-auto"
    >
      <h3
        class="margin-bottom-0 margin-top-5"
      >
        Test result information
      </h3>
      <p
        class="margin-bottom-0"
      >
        Please select test order and specimen type above to see test result information
      </p>
    </div>
  </div>
</div>
`;
